### ISSUE TITLE
Invalid python code in the documentation for ASYNC100 rule

### ISSUE THREAD

**Issue by olutra** [astral-sh/ruff#12921]
Examples for the rule **cancel-scope-no-checkpoint (ASYNC100)** are not valid python code
https://github.com/astral-sh/ruff/blob/d8debb7a36bad602b543a299f5cdca5b4c29524f/crates/ruff_linter/src/rules/flake8_async/rules/cancel_scope_no_checkpoint.rs#L29-L35

The code 
```python
import asyncio

async def func():
    with asyncio.timeout(2):
        do_something()
        await awaitable()

asyncio.run(func())
```
will throw a TypeError
`TypeError: 'Timeout' object does not support the context manager protocol`
because `asyncio.timeout` is an _asynchronous_ context manager.

**Comment by MichaReiser** [comment]
Thanks
